src/Compilers/CSharp/Test/Semantic/Semantics/RefEscapingTests.cs

```diff
@@ -8630,6 +8630,62 @@ static R F()
                 Diagnostic(ErrorCode.ERR_EscapeVariable, "y").WithArguments("y").WithLocation(25, 20));
         }
 
+        [Theory, CombinatorialData, WorkItem("https://github.com/dotnet/roslyn/issues/65353")]
+        public void Local_NestedLambda([CombinatorialRange(0, 3)] int nesting)
+        {
+            var source = $$"""
+                {{new string('{', nesting)}}
+                    int x = 1;
+                    F(() =>
+                    {
+                        int y = 2;
+                        ref int r = ref y;
+                        r = ref x;
+                    });
+                {{new string('}', nesting)}}
+
+                static void F(System.Action a) { }
+                """;
+            CreateCompilation(source).VerifyDiagnostics();
+        }
+
+        [Theory, CombinatorialData, WorkItem("https://github.com/dotnet/roslyn/issues/65353")]
+        public void Local_NestedLocalFunction([CombinatorialRange(0, 3)] int nesting)
+        {
+            var source = $$"""
+                {{new string('{', nesting)}}
+                    int x = 1;
+                    F();
+                    void F()
+                    {
+                        int y = 2;
+                        ref int r = ref y;
+                        r = ref x;
+                    }
+                {{new string('}', nesting)}}
+                """;
+            CreateCompilation(source).VerifyDiagnostics();
+        }
+
+        [Theory, CombinatorialData, WorkItem("https://github.com/dotnet/roslyn/issues/65353")]
+        public void Local_NestedLocalFunction_Parameter([CombinatorialRange(0, 3)] int nesting)
+        {
+            var source = $$"""
+                {{new string('{', nesting)}}
+                    int x = 1;
+                    F(ref x);
+                    void F(ref int z)
+                    {
+                        z = ref x;
+                    }
+                {{new string('}', nesting)}}
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (6,9): error CS8374: Cannot ref-assign 'x' to 'z' because 'x' has a narrower escape scope than 'z'.
```

Comment:
Something about this is not clicking for me. I think it's the fact that passing `ref x` as an argument is very similar to assigning `z = ref x`. Yet one is permitted and the other is not.

However, this seems like a pre-existing situation, which is not being changed by this PR.

It might be good to show what happens when using `scoped ref int z` and various levels of nesting.

Reply:
https://github.com/dotnet/roslyn/pull/75627#discussion_r1819874912